### DISCUSSION TITLE
How to create CRFPP docker container for armv7

### DISCUSSION THREAD

**Discussion by jleboula** [mealie-recipes/mealie#2052] [Q&A] [answered]
Hello!

I'm using mealie for a while now, and I love it, thanks for the good work here!

I'm still on a pretty old version due to the drop of armv7 support but I found my way in to setup something on v0.55 (if I remember correctly) on my RPI 4 on Raspbian 32bits.

Right now, I'm trying to do the same with v1.0.0beta-5 which is working so far until the CRFPP docker image pulling as it's not available for armv7 :/

Google is telling me that it might be a CRF++ toolkit implementation, but the docker image description dont provide enough data to tell me which one and how to redo it myself.
I'm seeing in an old post that https://github.com/mtlynch/ingredient-phrase-tagger might be the target but I cant confirm for sure (https://github.com/hay-kot/mealie/discussions/694#discussioncomment-1385067)

Is it possible to have more information about that? Like which library is used, have you made some specific changes to it, does it require a specific setup that is not described in the container description?

Thanks!

**Comment by michael-genson** [comment, marked as answer]
You can check here for details:

https://taku910.github.io/crfpp/


The parser is only important for parsing ingredients automatically, which you can get away with doing manually, so if there's a way to just skip the CRF++ package you might be okay with just doing that

When doing development we usually just skip it, unless specifically working on the parser

**Comment by jleboula** [comment]
Got it working!

Pretty straight forward in the end, I found the download link and created a container pulling the archive, and building it.
Now I have to deal with database migration as the format seems incompatible :D But that's another story!

Thanks a lot Michael!
